.github/workflows/test.yml

```diff
@@ -0,0 +1,160 @@
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/openassetio/openassetio-build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        uses: ./.github/build_openassetio_mediacreation
+
+  test-no-submodule:
+    name: Test no submodule
+    runs-on: ubuntu-latest
+    container:
+      image: aswf/ci-vfxall:2022-clang14.3
+    needs: [build-openassetio, build-openassetio-mediacreation]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get OpenAssetIO
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO Build
+          path: ./openassetio-build
+
+      - name: Get OpenAssetIO-MediaCreation
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO MediaCreation Build
+          path: ./openassetio-mediacreation-build
+
+      - name: Configure CMake build
+        run: >
+          cmake -S .
+          -DCMAKE_PREFIX_PATH="`pwd`/openassetio-build;`pwd`/openassetio-mediacreation-build/cmake/OpenAssetIO-MediaCreation"
+          -B build -G Ninja --preset=no-submodule
+
+      - name: Build tests
+        run: cmake --build build
+
+      - name: Test
+        run: ctest -VV --test-dir build
+
+  test-submodule-openassetio-find-mediacreation:
+    name: Test submodule openassetio find mediacreation
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/openassetio/openassetio-build
+    needs: [build-openassetio-mediacreation]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get OpenAssetIO-MediaCreation
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO MediaCreation Build
+          path: ./openassetio-mediacreation-build
+
+      - name: "Get Submodules"
+        run: git submodule update --init --recursive
+
+      - name: Configure CMake build
+        run: >
+          cmake -S .
+          -DCMAKE_PREFIX_PATH="`pwd`/openassetio-mediacreation-build/cmake/OpenAssetIO-MediaCreation"
+          -DOPENASSETIOTEST_ENABLE_PYTHON="OFF"
+          -DOPENASSETIO_ENABLE_C="ON"
+          -B build -G Ninja --preset=submodule-openassetio-find-mediacreation
+
+      - name: Build tests
+        run: cmake --build build
+
+      - name: Test
+        run: ctest -VV --test-dir build
+
+  test-find-openassetio-submodule-mediacreation:
+    name: Test find openassetio  submodule mediacreation
+    runs-on: ubuntu-latest
+    container:
+      image: aswf/ci-vfxall:2022-clang14.3
+    needs: [build-openassetio]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get OpenAssetIO
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO Build
+          path: ./openassetio-build
+
+      - name: Install Traitgen
+        run: |
+          python -m pip install git+https://github.com/feltech/OpenAssetIO-TraitGen@work/11-cppCodeGen
+
+      - name: "Get Submodules"
+        run: git submodule update --init --recursive
```

Comment:
[consider] Perhaps worth only initialising the submodule we're interested in?  I guess similar arguments to my comment re. `CMAKE_PREFIX_PATH`.

Reply:
Agreed with this, it keeps the files on disk minimal, which I think is important, not for performance, but to keep the build environment clean to avoid potential unknown accidental convergences in the future.
https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/7/commits/a2d521bb1098d443f0a71c915036a84de85c564a